src/extract/typed_header.rs

```diff
@@ -79,12 +79,14 @@ impl<T> Deref for TypedHeader<T> {
 #[cfg_attr(docsrs, doc(cfg(feature = "headers")))]
 #[derive(Debug)]
 pub struct TypedHeaderRejection {
-    name: &'static http::header::HeaderName,
-    reason: Reason,
+    /// Name of the header that caused the rejection
+    pub name: &'static http::header::HeaderName,
+    /// Reason why the header extraction has failed
+    pub reason: Reason,
 }
 
 #[derive(Debug)]
-enum Reason {
+pub enum Reason {
```

Comment:
The `typed_header` module isn't public so I don't think this type will actually be accessible. I think we should rename it to `TypedHeaderRejectionReason` it expose it in `axum::extract::rejection`, like the other rejection types.